src/main/java/insat/company/platform/service/impl/ClubServiceImpl.java

```diff
@@ -111,4 +130,70 @@ public void delete(Long id) {
             .stream(clubSearchRepository.search(queryStringQuery(query)).spliterator(), false)
             .collect(Collectors.toList());
     }
+
+    @Override
+    public JoinClubRequest sendClubJoinRequest(Club club, User user) {
+        if (!(club.hasMember(user))) {
+            return Optional.ofNullable(joinClubRequestRepository.findOneByUserAndClubAndStatus(user, club, Status.PENDING))
+                .map((obj) -> {
+                    if (obj.isPresent()) {
```

Comment:
I didnt understand why are you doing this?

Reply:
if there's already an existing Request pending to  join the same club , the user shouldn't  be able to send one . a null is returned tso that in the clubServiceImpl we can return a BAD_REQUEST .